src/test/test_pidfile.cc

```diff
@@ -0,0 +1,51 @@
+#include "global/pidfile.h"
+#include "common/ceph_argparse.h"
+#include "common/config.h"
+#include "include/cephfs/libcephfs.h"
+#include "include/rados/librados.h"
+#include "test/unit.h"
+
+#include <errno.h>
+#include <sstream>
+#include <string>
+#include <string.h>
+
+#include <boost/lexical_cast.hpp>
+using std::string
+pidfile_path[] = "test.pid";
+
+TEST(Pidfile, NoPidfile) {
+  md_config_t conf;
+  conf.pid_file[0] = '\0'; 
+  int ret;
+  ret = pidfile_open(&conf);
+  ASSERT_EQ(ret, 0);
+  ret = pidfile_write();
+  ASSERT_EQ(ret, 0);
+  ret = pidfile_remove();
+  ASSERT_EQ(ret, 0);
+}
+
+TEST(Pidfile, normal) {
+  md_config_t conf;
+  conf.pid_file = pidfile_path;
+  int ret;
+  ret = pidfile_open(&conf);
+  ASSERT_EQ(ret, 0);
+  ret = pidfile_write();
+  ASSERT_EQ(ret, 0);
+  ret = pidfile_remove();
+  ASSERT_EQ(ret, 0);
+}
+
+TEST(Pidfile, commendOneFile) {
```

Comment:
what's the diff between this test and the previous one? you meant to test the case where a second process is trying to open the same pidfile, i guess?